libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -26,7 +26,7 @@
 #include <libsolidity/ast/TypeProvider.h>
 #include <libsolidity/analysis/TypeChecker.h>
 #include <liblangutil/ErrorReporter.h>
-#include <boost/range/adaptor/reversed.hpp>
+#include <range/v3/view/reverse.hpp>
```

Comment:
You could add `using namespace ranges;` just like we do with `std` to make expressions shorter.

Reply:
I think though it may be nicer having an explicit `ranges` since it is non-std/non-solidity? The same way as we have explicit names for boost.

Reply:
Well, I'm not totally against having it explicit but I have to admit that even the `boost::` names are often too long and I'm often tempted to shorten them with something like `using fs = boost::filesystem`. But yeah, you're right that using the fully qualified name is the convention we use most often.

As an aside, personally I'd prefer to just import every name (even the ones from `std`) explicitly i.e. `using ranges::views::reverse;` because then they are short and it is still very easy to find out where each name comes from. This is how it is typically done in Python and I like this style a lot. We're not usually doing that in Solidity though. But I've seen it in @christianparpart's code so there's hope :P.

Reply:
Not sure why length is a problem. They are not used crazy many times, so clarity is nicer.

Reply:
so do we use namespaces or explicit ranges::

Reply:
If it was up to me I'd say drop it (that's what I've been doing in my code so far) but @axic has a point so in that case I'll accept it either way. You can just leave it explicit.